akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/CacheDirective.scala

```diff
@@ -86,22 +86,25 @@ object CacheDirectives {
   case object `must-revalidate` extends SingletonValueRenderable with ResponseDirective
 
   // http://tools.ietf.org/html/rfc7234#section-5.2.2.2
-  final case class `no-cache`(fieldNames: immutable.Seq[String]) extends FieldNamesDirective with ResponseDirective
+  final case class `no-cache`(fieldNames: colImmutable.Seq[String]) extends FieldNamesDirective with ResponseDirective
 
   // http://tools.ietf.org/html/rfc7234#section-5.2.2.5
   case object `public` extends SingletonValueRenderable with ResponseDirective
 
+  // https://tools.ietf.org/wg/httpbis/draft-ietf-httpbis-immutable/
+  case object `immutable` extends SingletonValueRenderable with ResponseDirective
+
   /** Java API */
   def getPublic: ResponseDirective = `public`
 
   // http://tools.ietf.org/html/rfc7234#section-5.2.2.6
-  final case class `private`(fieldNames: immutable.Seq[String]) extends FieldNamesDirective with ResponseDirective
+  final case class `private`(fieldNames: colImmutable.Seq[String]) extends FieldNamesDirective with ResponseDirective
```

Comment:
Hm, we must be careful with this change. 
Think about users of this class, now when they upgrade and they used the popular style of writing `immutable.Seq` etc all their code is broken (this change is not source compatible in a popular case for users).

I think we perhaps have to namespace the object somehow else...?